hub/constants.py

```diff
@@ -1,4 +1,17 @@
 BYTE_PADDING = b"\0"
+
+# number of bytes per unit
+KB = 1024
+MB = 1024 * KB
+GB = 1024 * MB
+
+MIN_MEMORY_CACHE_SIZE = 32 * MB
+MIN_LOCAL_CACHE_SIZE = 160 * MB
+
 CHUNKS_FOLDER = "chunks"
 META_FILENAME = "meta.json"
 INDEX_MAP_FILENAME = "index_map.json"
+
+PYTEST_MEMORY_PROVIDER_BASE_ROOT = "PYTEST_TMPDIR/memory_storage_provider/"
+PYTEST_LOCAL_PROVIDER_BASE_ROOT = "PYTEST_TMPDIR/local_storage_provider/"
```

Comment:
This looks like a `username/dataset` path. We either want `/tmp/PYTEST_TMPDIR` or `~/.tmp/PYTEST_TMPDIR` or something.

https://docs.pytest.org/en/6.2.x/tmpdir.html#the-default-base-temporary-directory
There might be a way to hook this up within pytest, but I'm not sure it's right for us